src/frontends/onnx/frontend/src/op/reduce.cpp

```diff
@@ -90,74 +90,98 @@ std::shared_ptr<ov::Node> get_reduction_axes_from_attr(const Node& node) {
     return v0::Constant::create(ov::element::i64, ov::Shape{reduction_axes.size()}, reduction_axes);
 }
 
+const std::set<element::Type> supported_types_v1 =
+    {element::u32, element::u64, element::i32, element::i64, element::f16, element::f32, element::f64};
+const std::set<element::Type> supported_types_v2 =
+    {element::u32, element::u64, element::i32, element::i64, element::f16, element::f32, element::f64, element::bf16};
```

Comment:
I think we do need to check input types to `Reduce` operation. That is because we assume input model is originally consistent (it comes with correct input type) and our `Reduce` operation should work for all of this operations

Reply:
This way we can explore limitation in core part and request to extend on new types if needed.
Let us say if some type not supported by core (or plugin, etc.) is needed by some model, then we should insert Convert operation that is not good.
We should rather to avoid any limitations in our part, assume that input model is consistent, and encourage other components avoid limitations.